### PR TITLE
Adding wso2 choreo support

### PR DESCRIPTION
**Pull request by theetaz**

### CODE REVIEW

File: infra/docker-evm-blockhain/Dockerfile [hunk 1]
```diff
@@ -3,18 +3,47 @@ FROM ethereum/client-go:alltools-latest
...
+COPY root-password.txt /app/root-password.txt
 
 # Copy the startup script
-COPY start.sh /root/start.sh
+COPY start.sh /app/start.sh
+
+# Set the proper permissions for the script
+RUN chmod -R 777 /app
+RUN chmod -R 777 /bin/sh
+RUN chmod -R 777 /bin/busybox
```
Comment (jdchathuranga) [inline infra/docker-evm-blockhain/Dockerfile:19]: Do we need 777 permission here? AFAIR we added 777 permission during the debug session, but do we really need 777 permission here?

File: infra/docker-evm-blockhain/Dockerfile [hunk 2]
```diff
@@ -3,18 +3,47 @@ FROM ethereum/client-go:alltools-latest
...
+
+# Set the proper permissions for the script
+# RUN chmod +x /app/start.sh
```
Comment (jdchathuranga) [inline infra/docker-evm-blockhain/Dockerfile:22]:
```suggestion
```

Remove un-necessary comment

File: infra/docker-evm-blockhain/Dockerfile [hunk 3]
```diff
@@ -3,18 +3,47 @@ FROM ethereum/client-go:alltools-latest
...
+
+RUN ls -la /app
+RUN ls -la /bin/sh
+RUN ls -la /bin/busybox
```
Comment (jdchathuranga) [inline infra/docker-evm-blockhain/Dockerfile:26]:
```suggestion
```

I think these `ls` are for debugging purposes.

File: infra/docker-evm-blockhain/Dockerfile [hunk 4]
```diff
@@ -3,18 +3,47 @@ FROM ethereum/client-go:alltools-latest
...
+
+# Create a user with a known UID/GID within range 10000-20000.
+# This is required by Choreo to run the container as a non-root user.
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --uid 10014 \
+    "choreo"
+
+RUN chown -R choreo:choreo /app
```
Comment (jdchathuranga) [inline infra/docker-evm-blockhain/Dockerfile:36]:
```suggestion
```

AFAIR we have added this `chown` for debugging purposes.  let's keep this only if it is really necessary.

File: infra/docker-evm-blockhain/Dockerfile [hunk 5]
```diff
@@ -3,18 +3,47 @@ FROM ethereum/client-go:alltools-latest
...
+
+# Use the above created unprivileged user
+USER 10014
+
+
 
 # Expose the required ports
 EXPOSE 30301 30310 30311 8040 8041 8551 8552
 
+RUN ls -la /app
```
Comment (jdchathuranga) [inline infra/docker-evm-blockhain/Dockerfile:46]:
```suggestion
```

Remove debug `ls`

File: infra/docker-evm-blockhain/Dockerfile (hunk 4 above)
Comment (theetaz) [reply infra/docker-evm-blockhain/Dockerfile]: Tested without running `chown -R choreo:choreo /app` command. container working as expected.

**Review by jdchathuranga — APPROVED** [review]
Looks good to me

**Review by yasithnd — APPROVED** [review]

**Review by jdchathuranga — APPROVED** [review]

**Pull request merged**